---
layout: post

title: "[BioInfomatics] BUSCO Installation & Usage"

tags:

- software
- bioinfomatics

category:
- software
- bioinfomatics


---
* toc
{:toc}


# BUSCO v3 Installation & Usage

BUSCO *v3* is a tool providing quantitative measures for the assessment of genome assembly, gene set, and transcriptome completeness, based on evolutionarily-informed expectations of gene content from near-universal single-copy orthologs selected from [OrthoDB *v9*](http://orthodb.org/).

BUSCO also provides user guide : [BUSCO_v3_userguide.pdf](https://gitlab.com/ezlab/busco/blob/master/BUSCO_v3_userguide.pdf)

cite : https://busco.ezlab.org



## Installation

I executed BUSCO v3 on **Ubuntu Server 16.04 LTS**. (Amazon AWS)

> 1. git clone https://gitlab.com/ezlab/busco.git
> 2. move directory to busco.
> 3. sudo python setup.py install.

Installation of BUSCO only is finished.

But, it needs additional software to use BUSCO v3 :(   (not described in detail in user guide)

Those are NCBI BLAST+ , HMMER , Augustus.

![FPTraffic](https://i-h1.pinimg.com/564x/6c/99/a0/6c99a0b8d272be67d80c5230ebadae61.jpg)



Especially, installation of Augustus is complex. here is a list of dependency to Augustus.

> ### bamtools, samtools, bcftools, htslib , boost, zlib1g 

And, each tool also have a dependency.

> #### bamtools - zlib 
>
> #### htslib - zlib, liblzma, libbzip2
>
> #### samtools - libncurses5-dev libncursesw5-dev

After installing all dependency to Augustus, should change path of software.

> Augustus/auxprogs/bam2hints/Makefile
>
> Augustus/auxprogs/filterBam/src/Makefile
>
> Augustus/auxprogs/bam2wig/Makefile

Then, you can run Augustus perfectly. 



```
I'll post easy-to-use install script soon.
```





# Usage

```
python BUSCO.py -i [SEQUENCE_FILE] -l [LINEAGE] -o [OUTPUT_NAME] -m [MODE] -sp [SPECIES] (-c [CPU count])

 -i FASTA FILE, --in 			FASTA FILE
 -l LINEAGE,    --lineage_path  Specify location of the BUSCO lineage data to be used.
 -o OUTPUT,     --out OUTPUT
 -m MODE,       --mode MODE     Specify which BUSCO analysis mode to run.
                                There are three valid modes:
                                - geno or genome, for genome assemblies 
                                - tran or transcriptome, for transcriptome assemblies 
                                - prot or proteins, for annotated gene sets (protein) 
 -sp SPECIES,   --species SPECIES  
                                Name of existing Augustus species gene finding 
                                parameters. See Augustus documentation for available 
                                options.
 -c N,          --cpu N         Specify the number (N=integer) of threads/cores to use.
```

